### PR TITLE
Replace `table` with `va-table` web component

### PR DESCRIPTION
**Pull request by nickjg231**
## Summary
- Replace `<table>` with `<va-table>` in the `table.drupal.liquid` file.

Tested manually using the page found at `/housing-assistance/home-loans/funding-fee-and-closing-costs/`

Some things to note: 
- va-table applies its own classes, so some of the columns are right-aligned as you'll see in the screenshots.

## Related issue(s)
closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2217

## Testing done
- Manually tested in Firefox

## Screenshots
<details><summary>Desktop screen changes</summary>
before: 

image

after:

image

</details>

<details><summary>Mobile screen changes</summary>
before:

image
image

after:

image
image

</details>

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria
- [x] Places with `<table>` are replaced properly with `<va-table>` and `<va-table-row>`
- [x] Updated files still work with web-components

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: script/github-actions/code-coverage-format-report.js [hunk 1]
```diff
@@ -11,30 +11,31 @@ const regex = /([^A-Za-z0-9/()%-])+\s/g;
 
 const codeCoverageData = codeCoverageReport.replace(regex, ',').split(',');
 
-let codeCoverageHTML = '<table> \n <thead> \n <tr> \n'; // html format
+let codeCoverageHTML =
+  '<va-table> \n <va-table-row slot="headers" key="header"> \n'; // html format
```
Comment (tjheffner) [inline script/github-actions/code-coverage-format-report.js:15]: I'm not sure we want this particular instance of `<table>` updated, as it is isn't live on the website.

This script gets used in the CI workflow
https://github.com/department-of-veterans-affairs/content-build/blob/main/.github/workflows/continuous-integration.yml#L278

Comment (micahchiang) [reply script/github-actions/code-coverage-format-report.js:15]: Thanks for the insight, will update!

File: src/site/paragraphs/table.drupal.liquid [hunk 2]
```diff
@@ -26,47 +26,27 @@
...
-<table data-template="paragraphs/table" data-entity-id="{{ entity.entityId }}" class="va-table va-table--responsive" role="table">
-        {% if entity.fieldTable.caption %}
-            <caption>{{ entity.fieldTable.caption }}</caption>
+<va-table data-template="paragraphs/table" data-entity-id="{{ entity.entityId }}" role="table">
+    {% assign colLabels = entity.fieldTable.value.0  %}
+    {% for value in entity.fieldTable.value %}
+        {% if forloop.first == true %}
+            <va-table-row slot="headers" key="header">
+                {% for column in value %}
+                    {% if column == "" or column == nil %}
```
Comment (micahchiang) [inline src/site/paragraphs/table.drupal.liquid:35]: I'm not sure how, but `---` was being rendered to fields that came through as empty or null, so I've added a check here and just render a span with some whitespace if the value is null.

Comment (jamigibbs) [reply src/site/paragraphs/table.drupal.liquid:35]: This is a default behavior of the web component for handling missing data: https://design.va.gov/storybook/?path=/docs/components-va-table--missing-data

File: script/github-actions/code-coverage-format-report.js (hunk 1 above)
Comment (jamigibbs) [reply script/github-actions/code-coverage-format-report.js]: I think we might have missed reverting back to `table` for this so I added it back here: https://github.com/department-of-veterans-affairs/content-build/pull/1817/commits/3b16534ad2a9f7b754eb690cf006021a9dc8ce46

**Comment by tjheffner** [comment]
The code changes LGTM, just want to confirm that the visual changes are expected / desired. No more borders around non-table-header cells, and the text-alignment feels different

Screenshot 2023-12-18 at 8 15 46 AM

The lack of borders appears consistent with: https://design.va.gov/storybook/?path=/docs/components-va-table--missing-data but I'm not as sure about the alignment?

**Comment by jamigibbs** [comment]
> The lack of borders appears consistent with: https://design.va.gov/storybook/?path=/docs/components-va-table--missing-data but I'm not as sure about the alignment?

@tjheffner Thanks for the review! Yes, the alignment is expected with the web component. We have an outstanding issue in our backlog to clarify this in our documentation: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1871

**Review by tjheffner — APPROVED** [review]

**Review by ataker — APPROVED** [review]

**Pull request merged**